### PR TITLE
Update context selector bar and show API endpoint URL

### PR DESCRIPTION
**Pull request by ppadti**
Closes: [RHOAIENG-15171](https://issues.redhat.com/browse/RHOAIENG-15171)

## Description
This PR aims to update the context bar selector to include help content and show the API endpoint URL of the selected model registry.

Screenshot 2024-11-14 at 1 18 28 PM

Screenshot 2024-11-14 at 1 18 10 PM

Screenshot 2024-11-14 at 3 58 23 PM

## How Has This Been Tested?
1. Go to Model registry
2. Click on `View details` button to see the API endpoint URL 
3. Click on `Need another registry` button to see help content.

## Test Impact
Added cypress test.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit or cypress tests for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by ppadti** [comment]
@yih-wang Could you please take a look at the screenshots?

**Comment by yih-wang** [comment]
Looks good, thanks @ppadti !
The only comment I have is for the indentation of the contact admin popover bullets - I've checked what we have for the MR empty state and feels like the bullets indentation should be smaller. See the screenshot below:
image

**Comment by ppadti** [comment]
@yih-wang updated the screenshots. Have a look when you have time.

**Comment by yih-wang** [comment]
LGTM, thanks @ppadti !

**Review by manaswinidas — APPROVED** [review]
/lgtm

Checked popover has the server URL with copy and the other popover is also present.

File: frontend/src/__mocks__/mockModelRegistryService.ts [hunk 1]
```diff
@@ -3,17 +3,26 @@ import { ServiceKind } from '~/k8sTypes';
...
   namespace?: string;
+  description?: string;
+  serverUrl?: string;
 };
 
 export const mockModelRegistryService = ({
   name = 'modelregistry-sample',
   namespace = 'odh-model-registries',
+  description = 'Model registry description',
+  serverUrl = 'modelregistry-sample-rest.apps.kserve-pm.dev.datahub.redhat.com:443',
```
Comment (manaswinidas) [inline frontend/src/__mocks__/mockModelRegistryService.ts:14]: oops missed this... small nit: this needn't be the same URL. Can this be something generic or are you validating this somewhere?

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelRegistry.cy.ts [hunk 2]
```diff
@@ -228,6 +232,34 @@ describe('Model Registry core', () => {
     modelRegistry.navigate();
     modelRegistry.shouldModelRegistrySelectorExist();
     modelRegistry.shouldregisteredModelsEmpty();
+
+    modelRegistry.findViewDetailsButton().click();
+    modelRegistry.findDetailsPopover().should('exist');
+    modelRegistry.findDetailsPopover().findByText('Model registry description').should('exist');
+    modelRegistry
+      .findDetailsPopover()
+      .findByText('https://modelregistry-sample-rest.apps.kserve-pm.dev.datahub.redhat.com:443')
```
Comment (manaswinidas) [inline frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelRegistry.cy.ts:241]: you have to change this too..

File: frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelRegistry.cy.ts [hunk 3]
```diff
@@ -228,6 +232,34 @@ describe('Model Registry core', () => {
...
+      .should('exist');
+
+    // Model registry with no description
+    modelRegistry.findModelRegistry().findSelectOption('modelregistry-sample-2').click();
+    modelRegistry.findViewDetailsButton().click();
+    modelRegistry.findDetailsPopover().should('exist');
+    modelRegistry.findDetailsPopover().findByText('No description').should('exist');
+    modelRegistry
+      .findDetailsPopover()
+      .findByText('https://modelregistry-sample-2-rest.apps.kserve-pm.dev.datahub.redhat.com:443')
```
Comment (manaswinidas) [inline frontend/src/__tests__/cypress/cypress/tests/mocked/modelRegistry/modelRegistry.cy.ts:251]: this too.

**Review by manaswinidas — CHANGES_REQUESTED** [review]

File: frontend/src/__mocks__/mockModelRegistryService.ts (hunk 1 above)
Comment (mturley) [reply frontend/src/__mocks__/mockModelRegistryService.ts:14]: @manaswinidas I'm not sure exactly what you mean here, are you saying this shouldn't be a specific real cluster URL and instead should be something like `'sample.url.com:443'`?

**Comment by mturley** [comment]
LGTM as well, just want to resolve the comments first

**Review by gitdallas — APPROVED** [review]

**Review by manaswinidas — APPROVED** [review]
/lgtm

**Pull request merged**